### PR TITLE
gatsby-theme-blog: Mdx -> BlogPost

### PR DESCRIPTION
**Pull request by ChristopherBiscardi**
## Description

Replace the queries for `Mdx` with queries for `BlogPost`. This provides a better potential for future compatibility with generic data interface work.

### CODE REVIEW

File: themes/gatsby-theme-blog/src/templates/post.js [hunk 1]
```diff
@@ -9,12 +9,10 @@ export default ({ pathContext: { previous, next }, location, data }) => (
 
 export const pageQuery = graphql`
   query($id: String!) {
-    post: mdx(id: { eq: $id }) {
+    post: blogPost(id: { eq: $id }) {
```
Comment (jxnblk) [inline themes/gatsby-theme-blog/src/templates/post.js]: This is cool

File: themes/gatsby-theme-blog/gatsby-node.js [hunk 2]
```diff
@@ -39,6 +40,55 @@ exports.onPreBootstrap = ({ store }, themeOptions) => {
...
+    schema.buildObjectType({
+      name: `BlogPost`,
+      fields: {
+        id: { type: `ID!` },
+        title: {
+          type: `String!`,
+        },
+        date: { type: `Date`, extensions: { dateformat: {} } },
+        tags: { type: `[String]!` },
+        keywords: { type: `[String]!` },
```
Comment (jxnblk) [inline themes/gatsby-theme-blog/gatsby-node.js]: Maybe a dumb question but what's the difference between *tags* and *keywords*?

Comment (ChristopherBiscardi) [reply themes/gatsby-theme-blog/gatsby-node.js]: keywords seems to be for SEO purposes while I'd expect tags to be user-facing. I'm open to removing this

**Comment by ChristopherBiscardi** [comment]
missing `slug: { type: 'String' }` in the `sourceNodes` type for `BlogPost`

File: themes/gatsby-theme-blog/gatsby-node.js (hunk 2 above)
Comment (amberleyromo) [reply themes/gatsby-theme-blog/gatsby-node.js]: ^yep. keywords are tag-oriented, for tags we could do tag filtering later.

**Pull request merged**